### PR TITLE
Support labels configuration for Github Pull Request

### LINKED ISSUE #108: support labels in updatecli values
For GitHub it would be nice to add labels on a pull request on creation.

```yaml
github:
  user: "GitHub Action"
  email: "41898282+github-actions[bot]@users.noreply.github.com"
  username: "github-actions"
  token: "UPDATECLI_GITHUB_TOKEN"
  branch: "master"
  owner: "owner"
  repository: "repo"
  labels:
    - dependency
```

If you like the idea, I would be open to working on the issue.
If not feel free to close the issue.

### PR DESCRIPTION
**Pull request by olblak**
# Support labels configuration for Github Pull Request
Fix #108

! This pull request introduces some refactoring to automatically re-opened a pull request in a closed.

Allow specifying a new block configuration "pullrequest" via updatecli config.

All settings in the pullRequest are optional!

**labels** - Allow specifying labels when creating a new pullrequest. 

! labels must already exist otherwise they are just 

```
      scm:
        github:
          user: "{{ .github.user }}"
          email: "{{ .github.email }}"
          owner: "{{ .github.owner }}"
          repository: "{{ .github.repository }}"
          token: "{{ requiredEnv .github.token }}"
          username: "{{ .github.username }}"
          branch: "{{ .github.branch }}"
          pullRequest:
            description: |
              Loprem ipsum
            labels:
              - documentation

```

## Test
To test this pull request, you can run the following commands:

```
  cp pkg/plugins/github/
  go test .
```

### Tradeoff

Considering that we can have different parameters to configure a new pullRequest, I decided to have a specific block for better clarity.

### Potential improvement

Allow specifying assignee/project

### CODE REVIEW

File: pkg/plugins/github/label_test.go [hunk 1]
```diff
@@ -7,6 +7,12 @@ import (
 
 func TestGetRepositoryLabelsInformation(t *testing.T) {
 
+	// Short mode also skip integration test that require credentials
```
Comment (dduportal) [inline pkg/plugins/github/label_test.go:10]: Good one! Reminds me I should do the opposite for all my unit tests...

File: pkg/plugins/github/label.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
+package github
+
+import (
+	"context"
+
+	"github.com/shurcooL/githubv4"
+	"github.com/sirupsen/logrus"
+)
+
+// repositoryLabel hold specific label information returned from Github API
```
Comment (dduportal) [inline pkg/plugins/github/label.go:10]:
```suggestion
// repositoryLabel holds specific label informations returned from Github API
```

File: pkg/plugins/github/label.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+								name
+								description
+							}
+							cursor
+						}
+					}
+				}
+			}
+
+	*/
```
Comment (dduportal) [inline pkg/plugins/github/label.go:49]:
```suggestion
```

File: pkg/plugins/github/label.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+											name
+											description
+										}
+										cursor
+									}
+		            }
+		          }
+						}
+					}
+	*/
```
Comment (dduportal) [inline pkg/plugins/github/label.go:151]:
```
```

**Review by dduportal — APPROVED** [review]
Sounds good ! Code review is ok for me (only nitpicking comments).

Haven't tried a lot of cases, let's see ho it behaves.

File: pkg/plugins/github/label.go (hunk 3 above)
Comment (olblak) [reply pkg/plugins/github/label.go:49]: I like to keep in the comment that snippet that I used in Github console for future iteration

**Comment by olblak** [comment]
> Sounds good ! Code review is ok for me (only nitpicking comments).

Thanks for reviewing, coming days will be pretty busy for me so I am planning to merge this PR after Wednesday

**Pull request merged**